tapfreighter/chain_porter.go

```diff
@@ -643,47 +643,16 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 	deliver := func(ctx context.Context, out TransferOutput) error {
 		key := out.ScriptKey.PubKey
 
-		// If this is an output that is going to our own node/wallet,
-		// we don't need to transfer the proof.
-		if out.ScriptKey.TweakedScriptKey != nil && out.ScriptKeyLocal {
-			log.Debugf("Not transferring proof for local output "+
-				"script key %x", key.SerializeCompressed())
-			return nil
-		}
-
-		// Un-spendable means this is a tombstone output resulting from
-		// a split.
-		unSpendable, err := out.ScriptKey.IsUnSpendable()
+		// We'll first check to see if the proof should be delivered.
+		shouldDeliverProof, err := out.ShouldDeliverProof()
 		if err != nil {
-			return fmt.Errorf("error checking if script key is "+
-				"unspendable: %w", err)
-		}
-		if unSpendable {
-			log.Debugf("Not transferring proof for un-spendable "+
-				"output script key %x",
-				key.SerializeCompressed())
-			return nil
-		}
-
-		// Burns are also always kept local and not sent to any
-		// receiver.
-		if len(out.WitnessData) > 0 && asset.IsBurnKey(
```

Comment:
Ah I see this was also just existing behavior, feel free to ignore those other comments.